components/brave_ads/core/browser/search_result_ad/test_web_page_util.cc

```diff
@@ -0,0 +1,122 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_ads/core/browser/search_result_ad/test_web_page_util.h"
+
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/string_piece.h"
+#include "third_party/blink/public/mojom/document_metadata/document_metadata.mojom.h"
+
+namespace brave_ads {
+
+namespace {
+
+schema_org::mojom::ValuesPtr CreateVectorValuesPtr(std::string value) {
+  return schema_org::mojom::Values::NewStringValues({std::move(value)});
+}
+
+schema_org::mojom::ValuesPtr CreateVectorValuesPtr(int64_t value) {
+  return schema_org::mojom::Values::NewLongValues({value});
+}
+
+class TestWebPageConstructor final {
+ public:
+  explicit TestWebPageConstructor(int attribute_index_to_skip)
+      : attribute_index_to_skip_(attribute_index_to_skip) {
+    web_page_ = CreateWebPage();
+  }
+  ~TestWebPageConstructor() = default;
+  TestWebPageConstructor(const TestWebPageConstructor&) = delete;
+  TestWebPageConstructor& operator=(const TestWebPageConstructor&) = delete;
+
+  blink::mojom::WebPagePtr GetTestWebPage() { return std::move(web_page_); }
+
+ private:
+  blink::mojom::WebPagePtr CreateWebPage() {
+    blink::mojom::WebPagePtr web_page = blink::mojom::WebPage::New();
+    schema_org::mojom::EntityPtr entity = schema_org::mojom::Entity::New();
+    entity->type = "Product";
+    schema_org::mojom::PropertyPtr property =
+        schema_org::mojom::Property::New();
+    property->name = "creatives";
+
+    std::vector<schema_org::mojom::EntityPtr> entity_values;
+    entity_values.push_back(CreateCreativeEntity());
+    property->values =
+        schema_org::mojom::Values::NewEntityValues(std::move(entity_values));
+
+    entity->properties.push_back(std::move(property));
+    web_page->entities.push_back(std::move(entity));
+
+    return web_page;
+  }
+
+  template <typename T>
+  void AddProperty(std::vector<schema_org::mojom::PropertyPtr>* properties,
+                   base::StringPiece name,
+                   T value) {
+    const int index = current_attribute_index_++;
+    if (index == attribute_index_to_skip_) {
+      return;
+    }
+    schema_org::mojom::PropertyPtr property =
+        schema_org::mojom::Property::New();
+    property->name = static_cast<std::string>(name);
+    property->values = CreateVectorValuesPtr(value);
+
+    properties->push_back(std::move(property));
+  }
+
+  schema_org::mojom::EntityPtr CreateCreativeEntity() {
+    const char* kSearchResultAdStringAttributes[] = {
```

Comment:
`constexpr const char*` to make clang-tidy happy and keep `kSearchResultAdStringAttributes` name.

Reply:
fixed